src/components/ContextMenu/ContextMenuItem.js

```diff
@@ -0,0 +1,127 @@
+import React, { useState } from 'react';
+import styled from 'styled-components';
+
+import ContextMenu from '.';
+
+function ContextMenuItem({
+  item,
+  parentRef,
+  isToLeft,
+  index,
+  checkedItem,
+  setCheckedItem,
+}) {
+  const [renderMenu, setRenderMenu] = useState(false);
+
+  function handleMouseEnter() {
+    setRenderMenu(true);
+  }
+
+  function handleMouseLeave() {
+    setRenderMenu(false);
+  }
+
+  function handleCheckedItem() {
+    setCheckedItem(index);
+  }
+
+  switch (item.type) {
+    case 'item':
+      return (
+        <StyledItem
+          className={item.inactive ? 'inactive' : ''}
+          showChecked={checkedItem === index}
+          onClick={handleCheckedItem}
+        >
+          {item.checked && (
+            <img className="checked" src={item.checked} alt="checked" />
+          )}
+          {item.icon && <img className="icon" src={item.icon} alt="icon" />}
+          {item.text}
+        </StyledItem>
+      );
+    case 'menu':
+      return (
+        <StyledItem
+          className="arrow"
+          onMouseEnter={handleMouseEnter}
+          onMouseLeave={handleMouseLeave}
+        >
+          {item.text}
+          {renderMenu && (
+            <ContextMenu
+              items={item.items}
+              parentRef={parentRef}
+              isToLeft={isToLeft}
+            />
+          )}
+        </StyledItem>
+      );
+    case 'separator':
+      return <StyledItem className="separator"></StyledItem>;
```

Comment:
This should be self closing - `<StyledItem className="separator" />`

Reply:
Fixed.

Reply:
Fixed.